zkevm-circuits/src/witness/tx.rs

```diff
@@ -48,97 +62,170 @@ pub struct Transaction {
-            vec![
-                [
-                    Value::known(F::from(self.id as u64)),
-                    Value::known(F::from(TxContextFieldTag::Nonce as u64)),
-                    Value::known(F::zero()),
-                    Value::known(F::from(self.nonce)),
-                ],
-                [
-                    Value::known(F::from(self.id as u64)),
-                    Value::known(F::from(TxContextFieldTag::Gas as u64)),
-                    Value::known(F::zero()),
-                    Value::known(F::from(self.gas)),
-                ],
-                [
-                    Value::known(F::from(self.id as u64)),
-                    Value::known(F::from(TxContextFieldTag::GasPrice as u64)),
-                    Value::known(F::zero()),
-                    challenges.evm_word().map(|evm_word| {
-                        RandomLinearCombination::random_linear_combine(
-                            self.gas_price.to_le_bytes(),
-                            evm_word,
-                        )
-                    }),
-                ],
-                [
-                    Value::known(F::from(self.id as u64)),
-                    Value::known(F::from(TxContextFieldTag::CallerAddress as u64)),
-                    Value::known(F::zero()),
-                    Value::known(self.caller_address.to_scalar().unwrap()),
-                ],
-                [
-                    Value::known(F::from(self.id as u64)),
-                    Value::known(F::from(TxContextFieldTag::CalleeAddress as u64)),
-                    Value::known(F::zero()),
-                    Value::known(self.callee_address.to_scalar().unwrap()),
-                ],
-                [
-                    Value::known(F::from(self.id as u64)),
-                    Value::known(F::from(TxContextFieldTag::IsCreate as u64)),
-                    Value::known(F::zero()),
-                    Value::known(F::from(self.is_create as u64)),
-                ],
-                [
-                    Value::known(F::from(self.id as u64)),
-                    Value::known(F::from(TxContextFieldTag::Value as u64)),
-                    Value::known(F::zero()),
-                    challenges.evm_word().map(|evm_word| {
-                        RandomLinearCombination::random_linear_combine(
-                            self.value.to_le_bytes(),
-                            evm_word,
-                        )
-                    }),
-                ],
-                [
-                    Value::known(F::from(self.id as u64)),
-                    Value::known(F::from(TxContextFieldTag::CallDataLength as u64)),
-                    Value::known(F::zero()),
-                    Value::known(F::from(self.call_data_length as u64)),
-                ],
+        let tx_hash_be_bytes = self.hash.to_fixed_bytes();
+        let tx_sign_hash_be_bytes = keccak256(&self.rlp_unsigned);
+        assert_eq!(self.hash.to_fixed_bytes(), keccak256(&self.rlp_signed));
```

Comment:
some mainnet blocks panic here.  If it is not a trivial bug and not easy to be fixed only by reading relative codes, i can give some commands to reproduce it later.(but may take some time)